bot/exts/evergreen/trivia_quiz.py

```diff
@@ -1,56 +1,236 @@
+    ans = random.randint(0, 9)  # max 9 because min mod 10
+    a = quotient * m + ans - b
+
+    question = q_format.format(a, b, m)
+    answer = a_format.format(ans)
+
+    return question, answer
+
+
+def ngonal_prism(q_format: str, a_format: str) -> Tuple[str, str]:
+    """Generate a question regarding vertices on n-gonal prisms."""
+    n = random.randint(0, len(N_PREFIXES) - 1)
+
+    question = q_format.format(N_PREFIXES[n])
+    answer = a_format.format((n + N_PREFIX_STARTS_AT) * 2)
+
+    return question, answer
+
+
+def imag_sqrt(q_format: str, a_format: str) -> Tuple[str, str]:
+    """Generate a negative square root question."""
+    ans_coeff = random.randint(3, 10)
+
+    question = q_format.format(ans_coeff ** 2)
+    answer = a_format.format(ans_coeff)
+
+    return question, answer
+
+
+def binary_calc(q_format: str, a_format: str) -> Tuple[str, str]:
+    """Generate a binary calculation question."""
+    a = random.randint(15, 20)
+    b = random.randint(10, a)
+    oper = random.choice(
+        (
+            ("+", operator.add),
+            ("-", operator.sub),
+            ("*", operator.mul),
+        )
+    )
+
+    if oper[0] == "*":
+        a -= 5
+        b -= 5
+
+    question = q_format.format(
+        a,
+        oper[0],
+        b,
+    )
+    answer = a_format.format(
+        oper[1](a, b)
+    )
+
+    return question, answer
+
+
+def solar_system(q_format: str, a_format: str) -> Tuple[str, str]:
+    """Generate a question on the planets of the Solar System."""
+    planet = random.choice(PLANETS)
+
+    question = q_format.format(planet[0])
+    answer = a_format.format(planet[1])
+
+    return question, answer
+
+
+def taxonomic_rank(q_format: str, a_format: str) -> Tuple[str, str]:
+    """Generate a question on taxonomic classification."""
+    level = random.randint(0, len(TAXONOMIC_HIERARCHY) - 2)
+
+    question = q_format.format(TAXONOMIC_HIERARCHY[level])
+    answer = a_format.format(TAXONOMIC_HIERARCHY[level + 1])
+
+    return question, answer
+
+
+def base_units_convert(q_format: str, a_format: str) -> Tuple[str, str]:
+    """Generate a SI base units conversion question."""
+    unit = random.choice(list(UNITS_TO_BASE_UNITS.keys()))
+
+    question = q_format.format(
+        unit + " " + UNITS_TO_BASE_UNITS[unit][0]
+    )
+    answer = a_format.format(
+        UNITS_TO_BASE_UNITS[unit][1]
+    )
+
+    return question, answer
+
+
+DYNAMIC_QUESTIONS_FORMAT_FUNCS = {
+    201: linear_system,
+    202: mod_arith,
+    203: ngonal_prism,
+    204: imag_sqrt,
+    205: binary_calc,
+    301: solar_system,
+    302: taxonomic_rank,
+    303: base_units_convert,
+}
+
 
 class TriviaQuiz(commands.Cog):
     """A cog for all quiz commands."""
 
     def __init__(self, bot: commands.Bot) -> None:
         self.bot = bot
-        self.questions = self.load_questions()
+
         self.game_status = {}  # A variable to store the game status: either running or not running.
```

Comment:
It's not immediately clear what's stored in these two dicts. What's the key and what's the value?
Perhaps add a type hint to explain that.
Same with `player_scores` and `game_player_scores` (why are those different? they have similar names) -- what is a score? an integer?

Reply:
i actually have no idea, all this was in the original code done by other people, i only added new stuff. so you'd have to ask them

Reply:
I completely understand and I would definitely add better documentation to the original code (made by others) if I could. However, even though I did do an analysis over the code before this project even started, I don't have the level of understanding of the previous quiz's code like its original programmers. Thus I cannot document it or else I might even mislead others. From what I understand, it's subtle differences of channels or something.

Reply:
Well, it's not very good that we have a piece of code nobody can understand. Code should be understandable by those who read it. If it's not, you should investigate and fix it.

From what I was able to gather from the code, `game_status` is a dict that maps a channel ID to a boolean flag. Its purpose is to make sure that at most one quiz is present in a channel, but to allow multiple quizzes in multiple channels.

So this is what I would do:
```py
self.is_game_running: Dict[int, bool] = {}  # maps channel ID to whether a quiz is running in that channel
```

Now, you also have two other variables:
```py
        self.game_owners = {} # A variable to store the person's ID who started the quiz game in a channel.
        self.player_scores = {}  # A variable to store all player's scores for a bot session.
        self.game_player_scores = {}  # A variable to store temporary game player's scores.
```
Do you see a pattern? All these 4 dicts take a channel ID and return a _game state_. So what you could do is put all that into one object:
```py
class Game:
    def __init__(self, owner_id: int):
        self.owner_id = owner_id
        self._scores: Dict[int, int] = {}

    def add_score(self, player_id: int, score_delta: int) -> None:
        self._scores[player_id] = self._scores.get(player_id, 0)  + score_delta

    def player_scores(self) -> Sequence[Tuple[int, int]]:
        """Returns a list of (user_id, score) tuples"""
        return list(self._scores.items())

```
inside the Cog class:
```py
self.games: Dict[int, Game] = {}  # maps channel ID to its current game
```

`self.player_scores` is actually not used at all (double-check, but it seems so), so remove it.